syntax = "proto3";

package grpc.auth;

// LoginRequest request.
message LoginRequest {
    string username = 1;
    string password = 2;
}

// LoginResponse response.
message LoginResponse {
    string user = 1;
    string jwt = 2;
    string error = 3;
}

// The Auth service definition.
service AuthService {
    // send a auth request
    rpc Login(LoginRequest) returns (LoginResponse) {}
}

// RegisterRequest request.
message RegisterRequest {
    string username = 1;
    string password = 2;
}

// RegisterResponse response.
message RegisterResponse {
    string user = 1;
    string error = 2;
}

// The Register service definition.
service RegisterService {
    // send a register request
    rpc Register(RegisterRequest) returns (RegisterResponse) {}
}
